cpp/ql/lib/semmle/code/cpp/models/implementations/Iterator.qll

```diff
@@ -131,16 +131,23 @@ private class IteratorPointerDereferenceOperator extends Operator, TaintFunction
 /**
  * A non-member `operator++` or `operator--` function for an iterator type.
  */
-private class IteratorCrementOperator extends Operator, DataFlowFunction {
+class IteratorCrementOperator extends Operator {
```

Comment:
The QLDoc for this class states that it only matches non-member iterator crement operators, whereas I suspect most users of this class will be looking for _any_ iterator crement operator whether its a member or not.  Should we have a public class that matches both, and perhaps suggest its use in the QLDoc for this class (and `IteratorCrementMemberOperator`)?  If there are no problems with this plan it might help people avoid the trap of using something overly specific because the class name sounded good.  We could even shuffle the names around (`IteratorCrementNonMemberOperator`) to make this even clearer.

The same goes for `IteratorAddOperator`, `IteratorSubOperator` and `IteratorAssignArithmeticOperator` if we decided to make any of these changes.

Reply:
I've renamed some of the classes in 7583fe2ad850773fd3939b36ff9292dabbf6a7c6 and added helpful QLDoc in daf7d1b7e7f24f48a0782a226dc047e83bd08840.